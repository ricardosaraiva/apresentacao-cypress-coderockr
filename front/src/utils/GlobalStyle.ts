import { createGlobalStyle } from 'styled-components';

export default createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;700&display=swap');

  body, *, input, textarea, select, button {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
    font-size: 16px;
    font-family: 'Roboto', sans-serif !important;
    outline: none !important;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  body {
    background-image: linear-gradient(
      to bottom right,
      ${({ theme }) => theme.palette.orange._500.toString()},
      ${({ theme }) => theme.palette.orange._400.toString()}
    );
    width: 100%;
    height: 100vh;
    color: ${({ theme }) => theme.palette.white.toString()};
  }
`;
