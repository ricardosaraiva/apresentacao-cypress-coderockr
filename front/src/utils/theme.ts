import 'styled-components';
import color from 'color';

const theme = {
  palette: {
    grey: {
      _50: color('#e3e7eb'),
      _100: color('#fafafa'),
      _700: color('#666'),
    },
    black: color('#000'),
    white: color('#FFF'),
    orange: {
      _400: color('#f7c974'),
      _500: color('#F2A91C'),
      _700: color('#af7303'),
    },
    green: {
      _500: color('#246224'),
    },
    red: {
      _500: color('#AA0601'),
    },
  },
};

type Theme = typeof theme

declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}

export default theme;
