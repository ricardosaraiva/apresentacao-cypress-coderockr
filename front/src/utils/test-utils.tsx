import { ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Router } from 'react-router-dom'
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { Model, Registry, Response, Server } from 'miragejs';

import theme from './theme';
import Schema from 'miragejs/orm/schema';
import { ModelDefinition } from 'miragejs/-types';

const history = createMemoryHistory();

type User = {
  id: number,
  name: string,
  email: string,
  password: string,
};

const UserModel: ModelDefinition<User> = Model.extend({});

type AppRegistry = Registry<{ users: typeof UserModel }, { }>
type AppSchema = Schema<AppRegistry>

const makeServer = (): Server => new Server({
  models: {
    users: Model,
  },
  routes: function() {
    this.timing = 100;
    this.urlPrefix = 'http://localhost:8080';
    this.logging = false;

    this.get('/user/me', async (schema: AppSchema, request) => {
      const user = await schema.find('users', request.queryParams.id);

      if (!user) {
        return new Response(400, {}, 'Usuário não encontrado');
      }

      return user.attrs;
    });

    this.post('/user/login', async (schema: AppSchema, request) => {
      const body = JSON.parse(request.requestBody);
      const user = await schema.findBy('users', {
        email: body.email,
        password: body.password,
      });

      if(!user) {
        return new Response(400, {}, 'Usuário não encontrado');
      }

      return user.attrs;
    });

    this.post('/user/create', async (schema: AppSchema, request) => {
      const body = JSON.parse(request.requestBody);

      const user = await schema.findBy('users', {
        email: body.email,
        password: body.password,
      });

      if(user) {
        return new Response(400, {}, 'Usuário duplicado');
      }

      await schema.create('users', {
        _id: 1,
        name: body.name,
        email: body.email,
        password: body.password,
      });

      const userData = await schema.find('users', '1');

      return userData?.attrs;
    });
  },
});

const renderSetup = (
  component: ReactNode,
  options?: Omit<RenderOptions, 'queries'>,
) => {
  return render (
    <ThemeProvider theme={theme}>
      <Router history={history}>
        {component}
      </Router>
    </ThemeProvider>,
    options,
  )
};

export * from '@testing-library/react'
export { render as renderWihtoutTheme};
export { renderSetup as render };
export { history };
export { makeServer };
