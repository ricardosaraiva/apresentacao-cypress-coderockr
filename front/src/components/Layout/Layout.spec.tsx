import { renderWihtoutTheme } from '../../utils/test-utils';

import Layout from './Layout';

describe('Componente Layout', () => {
    test('Valida se o componente layout esta renderizando', () => {
        const { container } = renderWihtoutTheme(<Layout>layouts</Layout>);

        const el = container.querySelector('div') as Element;

        expect(window.getComputedStyle(el)).toMatchSnapshot();
        expect(container).toMatchSnapshot();
    });
})
