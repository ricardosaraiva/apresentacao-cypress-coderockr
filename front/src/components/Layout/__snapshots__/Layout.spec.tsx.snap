// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Layout Valida se o componente layout esta renderizando 1`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "margin",
  "2": "visibility",
  "_importants": Object {
    "display": "",
    "margin": "",
    "visibility": undefined,
  },
  "_length": 3,
  "_onChange": [Function],
  "_values": Object {
    "display": "block",
    "margin": "0px 20px",
    "margin-bottom": "0px",
    "margin-left": "20px",
    "margin-right": "20px",
    "margin-top": "0px",
    "visibility": "visible",
  },
}
`;

exports[`Componente Layout Valida se o componente layout esta renderizando 2`] = `
<div>
  <div
    class="sc-bdfBwQ fmsyoN"
  >
    layouts
  </div>
</div>
`;
